src/hotspot/share/gc/shared/memAllocator.hpp

```diff
@@ -25,9 +25,9 @@
 #ifndef SHARE_GC_SHARED_MEMALLOCATOR_HPP
 #define SHARE_GC_SHARED_MEMALLOCATOR_HPP
 
-#include "gc/shared/collectedHeap.hpp"
 #include "memory/memRegion.hpp"
 #include "oops/oopsHierarchy.hpp"
+#include "runtime/thread.hpp"
```

Comment:
If we want to, this could be changed to a forward declaration if we removed the default value (Thread* thread = Thread::current()) of the constructors. Not needed for this RFE though.

Reply:
memAllocator.hpp is not included very often (only 65 out of ~1000 .o files), so I decided to leave it as is.